### DISCUSSION TITLE
Support for user added packages for use in Jupyterlite notebooks

### DISCUSSION THREAD

**Discussion by ashok777** [jupyterlite/jupyterlite#405] [Ideas]
I am very pleased with how Jupyterlite works and how it lends itself to being embedded in a host application.

It would really be beneficial for a notebook service provider  to have the ability to add packages for use in the pyolite notebooks. The ability to add 'requests', for instance, will allow the notebook user to make http requests conveniently. I am hoping the capability is in the works. 

Another issue: When importing pandas I get the following warning. It looks to be benign, and if so can it be suppressed?

**lib/python3.9/site-packages/pandas/compat/__init__.py:117: UserWarning: Could not import the lzma module. Your installed Python is incomplete. Attempting to use lzma compression will result in a RuntimeError**

**Comment by jtpio** [comment]
Thanks @ashok777 for raising these points.

There is ongoing work in https://github.com/jupyterlite/jupyterlite/pull/310 which should already help managing the packages.

For `requests` there is some discussion in https://github.com/pyodide/pyodide/issues/398. An alternative is to use `fetch` which is available from the Python kernel. There is an example in the `pyolite` notebook:

https://jupyterlite.readthedocs.io/en/latest/_static/lab?path=pyolite.ipynb

> Another issue: When importing pandas I get the following warning. It looks to be benign, and if so can it be suppressed?

This will probably be fixed upstream in Pyodide at some point. For now it is tracked in this issue: https://github.com/jupyterlite/jupyterlite/issues/273
